app/upgrades/v1rc7/testnet/upgrade.go

```diff
@@ -0,0 +1,101 @@
+package testnet
+
+import (
+	"context"
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/types/query"
+
+	"github.com/babylonlabs-io/babylon/app/keepers"
+	"github.com/babylonlabs-io/babylon/app/upgrades"
+	btcstakingkeeper "github.com/babylonlabs-io/babylon/x/btcstaking/keeper"
+	btcstakingtypes "github.com/babylonlabs-io/babylon/x/btcstaking/types"
+)
+
+const (
+	UpgradeName = "v1rc7"
+)
+
+func CreateUpgrade() upgrades.Upgrade {
+	return upgrades.Upgrade{
+		UpgradeName:          UpgradeName,
+		CreateUpgradeHandler: CreateUpgradeHandler(),
+	}
+}
+
+// CreateUpgradeHandler upgrade handler for launch.
+func CreateUpgradeHandler() upgrades.UpgradeHandlerCreator {
+	return func(mm *module.Manager, cfg module.Configurator, keepers *keepers.AppKeepers) upgradetypes.UpgradeHandler {
+		return func(context context.Context, _plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+			ctx := sdk.UnwrapSDKContext(context)
+
+			logger := ctx.Logger().With("upgrade", UpgradeName)
+			logger.Info("migrating finality providers...")
+			if err := MigrateFinalityProviders(ctx, keepers.BTCStakingKeeper); err != nil {
+				return nil, fmt.Errorf("failed migrate finality providers: %w", err)
+			}
+			logger.Info("finality providers migration done!")
+
+			migrations, err := mm.RunMigrations(ctx, cfg, fromVM)
+			if err != nil {
+				return nil, fmt.Errorf("failed to run migrations: %w", err)
+			}
+
+			return migrations, nil
+		}
+	}
+}
+
+// MigrateFinalityProviders populates the new CommissionInfo field
+// with default values
+func MigrateFinalityProviders(goCtx context.Context, k btcstakingkeeper.Keeper) error {
+	var (
+		defaultCommissionMaxRate       = sdkmath.LegacyMustNewDecFromStr("0.2")
+		defaultCommissionMaxChangeRate = sdkmath.LegacyMustNewDecFromStr("0.01")
+		ctx                            = sdk.UnwrapSDKContext(goCtx)
+		pagKey                         = []byte{}
+	)
+	for {
+		// FinalityProviders query is paginated, so we'll need to use
+		// the page key to make sure to migrate all of them
+		res, err := k.FinalityProviders(goCtx, &btcstakingtypes.QueryFinalityProvidersRequest{
+			Pagination: &query.PageRequest{Key: pagKey},
+		})
+		if err != nil {
+			return err
+		}
+
+		for _, fp := range res.FinalityProviders {
+			err := k.UpdateFinalityProvider(goCtx, &btcstakingtypes.FinalityProvider{
+				Addr:                 fp.Addr,
+				Description:          fp.Description,
+				Commission:           fp.Commission,
+				BtcPk:                fp.BtcPk,
+				Pop:                  fp.Pop,
+				SlashedBabylonHeight: fp.SlashedBabylonHeight,
+				SlashedBtcHeight:     fp.SlashedBtcHeight,
+				Jailed:               fp.Jailed,
+				HighestVotedHeight:   fp.HighestVotedHeight,
+				ConsumerId:           ctx.ChainID(),
```

Comment:
hmm this field will no be available in `release/v1.x` branch (as this is future feature) but this will need to be fixed during backport.

Here this field should be taken from `fp.ConsumerId` though it seems query is missing it 🤔 

Maybe, this upgrade won't affect the phase-3 in this shape so maybe for now lets:
- leave comments about this here
- fix this situation during backport, we will need to do it either way, as on `release/v1.x` branch this field just not exist

Reply:
@KonradStaniec got it. should we include the `ConsumerId` on the query response for the code on `main`? I can add that change if we want to include this field

Reply:
hmm we can do this as separate pr to `main` 👍